### PR TITLE
Support return from lambda (fixes opal#260)

### PR DESCRIPTION
**Pull request by hmdne**
The lambdas are now behaving exactly as they do in Ruby, they work just
like the def in terms of the return logic.

### CODE REVIEW

**Comment by hmdne** [comment]
I'm surprised there are no MSpec tests for this behavior...

File: lib/opal/nodes/scope.rb [hunk 1]
```diff
@@ -101,6 +101,24 @@ def def?
         @type == :def || @type == :defs
       end
 
+      def lambda?
+        iter? && @is_lambda
+      end
+
+      def is_lambda!
```
Comment (elia) [inline lib/opal/nodes/scope.rb:108]: The lint message can be fixed by append `# rubocop:disable Naming/PredicateName` to the `def` line

File: spec/filters/bugs/kernel.rb [hunk 2]
```diff
@@ -411,4 +411,5 @@
   fails "Kernel.sprintf returns a String in the same encoding as the format String if compatible" # NameError: uninitialized constant Encoding::KOI8_U
   fails "Kernel.sprintf width specifies the minimum number of characters that will be written to the result" # Expected "         1.095200e+02" to equal "        1.095200e+02"
   fails "Kernel.srand returns the previous seed value on the first call" # NoMethodError: undefined method `insert' for "rubyexe.rb"
+  fails "Kernel.lambda treats the block as a Proc when lambda is re-defined" # Expected 2 == 1 to be truthy but was false
```
Comment (elia) [inline spec/filters/bugs/kernel.rb:414]: Thoughts on supporting re-definition of `lambda` using a check on `$$pristine`?

**Review by elia — APPROVED** [review]
👍 

I left a suggestion on how to also support lambda redefinition, but this is more important and we can definitely merge and think about that later

File: spec/filters/bugs/kernel.rb (hunk 2 above)
Comment (hmdne) [reply spec/filters/bugs/kernel.rb:414]: Nope. This is a compiler-level feature, it doesn't know what runtime does.

Comment (hmdne) [reply spec/filters/bugs/kernel.rb:414]: Unless... we change the returner, but then a returner would need to be compiled also for the parent scope.

**Pull request merged**